DataGenerator/src/server.js

```diff
@@ -51,25 +38,27 @@ const server = net.createServer((socket) => {
     exit();
   });
 
+  // Pack and send a buffer at half second intervals
   interval = setInterval(() => {
     buffOffset = 0; // Offset when adding each value to buf1
-    nextValue = (nextValue + 1) % 100; // Generate a new value
 
     // Fill buf1 with new data according to the data format file
     for (const property in DATA_FORMAT) {
+      nextValue = Math.abs(Math.sin(nextValue)) * 100; // Generate a new value
+
       // Add the next value to the Buffer based on the data type
       switch (DATA_FORMAT[property][1]) {
         case "uint8":
-          buf1.writeUInt8(nextValue, buffOffset);
+          buf1.writeUInt8(Math.round(nextValue), buffOffset);
           break;
         case "float":
-          buf1.writeFloatBE(nextValue + 0.125, buffOffset);
+          buf1.writeFloatLE(Math.floor(nextValue) + 0.125, buffOffset);
           break;
         case "char":
-          buf1.writeUInt8(nextValue, buffOffset);
+          buf1.writeUInt8(Math.round(nextValue), buffOffset);
           break;
         case "bool":
-          buf1.writeUInt8(nextValue % 2, buffOffset);
+          buf1.writeUInt8(Math.round(nextValue) % 2, buffOffset);
```

Comment:
Why % 2? Can we put a comment or put it in a function?

Reply:
Added a comment